datalad/support/annexrepo.py

```diff
@@ -763,21 +763,44 @@ def is_special_annex_remote(self, remote, check_if_known=True):
     @borrowkwargs(GitRepo)
     def get_remotes(self,
                     with_urls_only=False,
-                    exclude_special_remotes=False):
+                    exclude_special_remotes=False,
+                    exclude_disabled=True):
         """Get known (special-) remotes of the repository
 
         Parameters
         ----------
         exclude_special_remotes: bool, optional
           if True, don't return annex special remotes
+        exclude_disabled: bool, optional
+          if True, don't query annex for not (yet) enabled (special) remotes
 
         Returns
         -------
         remotes : list of str
           List of names of the remotes
         """
+
         remotes = super(AnnexRepo, self).get_remotes(with_urls_only=with_urls_only)
 
+        # Note: Remotes with an url are already included in `remotes`, since by definition there's a
+        # remote.NAME.url config
+        if not exclude_disabled and not exclude_special_remotes and not with_urls_only:
+            annex_remotes = []
+            repo_info = self.repo_info(fast=True)
+            descriptions = [r["description"]
+                            for r in repo_info["untrusted repositories"] +
+                                     repo_info["semitrusted repositories"] +
+                                     repo_info["trusted repositories"]
+                            if not r["here"] and r["description"] not in ["web", "bittorrent"]]
+            for d in descriptions:
+                parts = d.split()
+                if len(parts) >= 2 and parts[-1].startswith('['):
+                    annex_remotes.append(parts[-1][1:-1])
+                else:
+                    annex_remotes.append(d)
```

Comment:
This is kind of just repeating the same "inaccessible regular remotes" comment I've already made, but I think using the description (so free form text) here could lead to confusing results, particularly because it is cross-referenced with `remotes` in the next line.

```
$ cd $(mktemp -d --tmpdir dl-XXXXXXX)
$ datalad create -D"this is a" a
$ datalad install -D"this is b" -s a b
$ git -C b annex sync
$ datalad install -D"this is c" -s a c
$ cd c
$ git annex info
repository mode: indirect
trusted repositories: 0
semitrusted repositories: 5
	00000000-0000-0000-0000-000000000001 -- web
 	00000000-0000-0000-0000-000000000002 -- bittorrent
 	053a6d9c-6dae-4c08-994e-e20fac621b10 -- this is b
 	1ef35224-2135-473f-82a2-f5a467449364 -- this is a [origin]
 	30742caf-85fb-4be9-86e5-ea331f85b439 -- this is c [here]
untrusted repositories: 0
transfers in progress: none
available local disk space: 238.22 gigabytes (+1 megabyte reserved)
local annex keys: 0
local annex size: 0 bytes
annexed files in working tree: 0
size of annexed files in working tree: 0 bytes
bloom filter size: 32 mebibytes (0% full)
backend usage: 
$ python -c "from datalad.support.annexrepo import AnnexRepo; print(AnnexRepo('.').get_remotes(exclude_disabled=False))" 
['origin', 'this is b']
```

Reply:
Well done, thanks!
IIRC the issue was, that `annex info --json` (and thereby `repo_info()`) doesn't provide the name in any case, so had to rely on the description field. But obviously that doesn't really work in those cases.
Not sure yet, what to do about it. Might turn out to be a serious show stopper.